runtime/peaq-dev/src/lib.rs

```diff
@@ -62,92 +83,25 @@ pub use frame_support::{
-use parachain_staking::reward_rate::RewardRateInfo;
-
-use pallet_ethereum::{Call::transact, Transaction as EthereumTransaction};
-use pallet_evm::{
-	Account as EVMAccount, EnsureAddressTruncated, GasWeightMapping, HashedAddressMapping, Runner,
-};
 pub use pallet_timestamp::Call as TimestampCall;
-
-#[cfg(any(feature = "std", test))]
-pub use sp_runtime::BuildStorage;
-
-mod precompiles;
-pub use precompiles::PeaqPrecompiles;
-pub type Precompiles = PeaqPrecompiles<Runtime>;
-
-// Polkadot imports
-use polkadot_runtime_common::{BlockHashCount, SlowAdjustingFeeUpdate};
-
-use cumulus_pallet_parachain_system::RelayNumberStrictlyIncreases;
-
-pub use peaq_primitives_xcm::*;
-use peaq_rpc_primitives_txpool::TxPoolResponse;
-
 pub use peaq_pallet_did;
-use peaq_pallet_did::{did::Did, structs::Attribute as DidAttribute};
+pub use peaq_pallet_mor::{self, types::MorConfig};
 pub use peaq_pallet_rbac;
-use peaq_pallet_rbac::{
-	error::RbacError,
-	rbac::{Group, Permission, Rbac, Result as RbacResult, Role},
-	structs::{
-		Entity as RbacEntity, Permission2Role as RbacPermission2Role, Role2Group as RbacRole2Group,
-		Role2User as RbacRole2User, User2Group as RbacUser2Group,
-	},
-};
 pub use peaq_pallet_storage;
-use peaq_pallet_storage::traits::Storage;
 pub use peaq_pallet_transaction;
+pub use precompiles::PeaqPrecompiles;
+#[cfg(any(feature = "std", test))]
+pub use sp_runtime::BuildStorage;
 
-use peaq_pallet_mor::mor::MorBalance;
-pub use peaq_pallet_mor::{self, types::MorConfig};
-
-// For XCM
-mod weights;
-pub mod xcm_config;
-use orml_currencies::BasicCurrencyAdapter;
-use orml_traits::parameter_type_with_key;
-pub mod constants;
-use xcm::latest::prelude::*;
-
-// For Zenlink-DEX-Module
-use zenlink_protocol::{AssetBalance, MultiAssetsHandler, PairInfo, ZenlinkMultiAssets};
-
-use runtime_common::{
-	CurrencyHooks, LocalAssetAdaptor, OperationalFeeMultiplier, PeaqCurrencyAdapter,
-	PeaqCurrencyPaymentConvert, TransactionByteFee, CENTS, DOLLARS, MILLICENTS,
-};
-
-/// An index to a block.
-type BlockNumber = peaq_primitives_xcm::BlockNumber;
-
-/// Alias to 512-bit hash when used in the context of a transaction signature on the chain.
-pub type Signature = peaq_primitives_xcm::Signature;
-
-/// Some way of identifying an account on the chain. We intentionally make it equivalent
-/// to the public key of our transaction signing scheme.
-pub type AccountId = peaq_primitives_xcm::AccountId;
-
-/// The type for looking up accounts. We don't expect more than 4 billion of them, but you
-/// never know...
-// type AccountIndex = peaq_primitives_xcm::AccountIndex;
-
-/// Index of a transaction in the chain.
-type Index = peaq_primitives_xcm::Nonce;
-
-/// A hash of some data used by the chain.
-type Hash = peaq_primitives_xcm::Hash;
-
-/// The ID of an entity (RBAC)
-type EntityId = [u8; 32];
+pub type Precompiles = PeaqPrecompiles<Runtime>;
+// Note: this is really wild! You can define it here, but not in peaq_primitives_xcm...?!
```

Comment:
Hahaha, we can move define here

Reply:
🤣 I just realised, that I was in operational changing mode (brain standby) - of course can this definition not be in `peaq-primitives-xcm`, because there is no Runtime available... but I will check, if we can just paste the `Precompile<Runtime>`, what do you think? Which style do you like more?

Reply:
Do you mean to use the `Precompiles<AgungRuntime>`, `Precompiles<KrestRuntime>` in the node folder? To me, that's also fine because no matter how we should compose the template. The only thing I'm thinking is whether it's easy to compose that in node folder

Reply:
Yes, I switched accidentially the topics... I was thinking about the Precompiles definition, which I also removed now, I like this style more... It is much cleaner and will only be used once in each chain-spec-file in node...
I was confused where this comment belonged to and changes this too... but I still don't understand why Block cannot be defined within peaq-primitives-xcm :thinking: There is no generic included for the `Runtime` definition... :thinking:

Reply:
Got it. According to our discuss, I think we can move it to the primitive-xcm if we solve the compilation issue, it'll be clearer